From 923e1968e093d36091faace0c38c43214aaeddd3 Mon Sep 17 00:00:00 2001
From: kitakar5525 <34676735+kitakar5525@users.noreply.github.com>
Date: Sat, 11 Apr 2020 11:33:35 +0000
Subject: [PATCH] Revert "FROMLIST: Input: atkbd: Receive and use
 physcode->keycode mapping from FW"

This reverts commit c29d72eb8a0919320de503ed339c07d8b1a8d7a0.

Reason for revert:
  seems to cause sensors not working (?)

Signed-off-by: Tsuchiya Yuto (kitakar5525) <kitakar@gmail.com>
---
 drivers/input/keyboard/atkbd.c | 41 +---------------------------------
 1 file changed, 1 insertion(+), 40 deletions(-)

diff --git a/drivers/input/keyboard/atkbd.c b/drivers/input/keyboard/atkbd.c
index 8de6330b6470..7fa066fb08f7 100644
--- a/drivers/input/keyboard/atkbd.c
+++ b/drivers/input/keyboard/atkbd.c
@@ -70,9 +70,6 @@ MODULE_PARM_DESC(terminal, "Enable break codes on an IBM Terminal keyboard conne
 
 #define MAX_FUNCTION_ROW_KEYS	24
 
-#define PHYSCODE(keymap)	((keymap >> 16) & 0xFFFF)
-#define KEYCODE(keymap)		(keymap & 0xFFFF)
-
 /*
  * Scancode to keycode tables. These are just the default setting, and
  * are loadable via a userland utility.
@@ -1039,38 +1036,6 @@ static unsigned int atkbd_oqo_01plus_scancode_fixup(struct atkbd *atkbd,
 	return code;
 }
 
-static int atkbd_get_keymap_from_fwnode(struct atkbd *atkbd)
-{
-	struct device *dev = &atkbd->ps2dev.serio->dev;
-	int i, n;
-	u32 *ptr;
-	u16 physcode, keycode;
-
-	/* Parse "linux,keymap" property */
-	n = device_property_count_u32(dev, "linux,keymap");
-	if (n <= 0 || n > ATKBD_KEYMAP_SIZE)
-		return -ENXIO;
-
-	ptr = kcalloc(n, sizeof(u32), GFP_KERNEL);
-	if (!ptr)
-		return -ENOMEM;
-
-	if (device_property_read_u32_array(dev, "linux,keymap", ptr, n)) {
-		dev_err(dev, "problem parsing FW keymap property\n");
-		kfree(ptr);
-		return -EINVAL;
-	}
-
-	memset(atkbd->keycode, 0, sizeof(atkbd->keycode));
-	for (i = 0; i < n; i++) {
-		physcode = PHYSCODE(ptr[i]);
-		keycode = KEYCODE(ptr[i]);
-		atkbd->keycode[physcode] = keycode;
-	}
-	kfree(ptr);
-	return 0;
-}
-
 /*
  * atkbd_set_keycode_table() initializes keyboard's keycode table
  * according to the selected scancode set
@@ -1078,16 +1043,13 @@ static int atkbd_get_keymap_from_fwnode(struct atkbd *atkbd)
 
 static void atkbd_set_keycode_table(struct atkbd *atkbd)
 {
-	struct device *dev = &atkbd->ps2dev.serio->dev;
 	unsigned int scancode;
 	int i, j;
 
 	memset(atkbd->keycode, 0, sizeof(atkbd->keycode));
 	bitmap_zero(atkbd->force_release_mask, ATKBD_KEYMAP_SIZE);
 
-	if (!atkbd_get_keymap_from_fwnode(atkbd)) {
-		dev_dbg(dev, "Using FW keymap\n");
-	} else if (atkbd->translated) {
+	if (atkbd->translated) {
 		for (i = 0; i < 128; i++) {
 			scancode = atkbd_unxlate_table[i];
 			atkbd->keycode[i] = atkbd_set2_keycode[scancode];
@@ -1215,7 +1177,6 @@ static void atkbd_parse_fwnode_data(struct serio *serio)
 		atkbd->num_function_row_keys = n;
 		dev_dbg(dev, "FW reported %d function-row key locations\n", n);
 	}
-
 }
 
 /*
-- 
2.17.1

